dataflow/src/main/java/org/checkerframework/dataflow/analysis/ConditionalTransferResult.java

```diff
@@ -113,12 +116,13 @@ public ConditionalTransferResult(
     this.thenStore = thenStore;
     this.elseStore = elseStore;
     this.storeChanged = storeChanged;
+    this.store = thenStore.leastUpperBound(elseStore);
```

Comment:
The then and else stores can change after the ConditionalTransferResult is created, so I don't think this will work.  I suspect that's why the tests aren't failing.  If you can get the tests to pass, I can take a closer look.

Reply:
Thanks for your comment. I will work on getting the tests to pass.

Reply:
@smillst There are a lot of places that store can change and I am not sure whether tracking those changes is more efficient. I closed this PR and opened a new [PR](https://github.com/typetools/checker-framework/pull/5107) that cache the result of getStoreAfter in MustCallConsistencyAnalyzer where stores can't change.